packages/cmrjs/tests/cmr-utils/test-reconcilecmrmetadata.js

```diff
@@ -5,31 +5,48 @@ const rewire = require('rewire');
 
 const cmrUtils = rewire('../../cmr-utils');
 
-const { BucketsConfig, log } = require('@cumulus/common');
+const { aws, BucketsConfig, log } = require('@cumulus/common');
 
 const { randomId } = require('@cumulus/common/test-utils');
 
 
-function setTestCredentials() {
-  process.env.cmr_provider = randomId('cmr_provider');
-  process.env.cmr_client_id = randomId('cmr_client_id');
-  process.env.cmr_username = randomId('cmr_username');
-  process.env.cmr_password = randomId('cmr_password');
-
+function getTestCredentials(cmrPassword) {
   return {
     provider: process.env.cmr_provider,
     clientId: process.env.cmr_client_id,
     username: process.env.cmr_username,
-    password: process.env.cmr_password
+    password: cmrPassword
   };
 }
 
+test.before(async (t) => {
+  // Store the CMR password
+  t.context.cmrPassword = randomId('cmr_password');
+  t.context.cmrPasswordSecretName = randomId('cmr_password_secret_name');
+  process.env.cmr_password_secret_name = t.context.cmrPasswordSecretName;
+  await aws.secretsManager().createSecret({
```

Comment:
At a quick glance, I didn't see any test that was passing the PW in directly. Can you link to a line in one the tests so I can see what I'm missing?

Reply:
https://github.com/nasa/cumulus/pull/1397/files#diff-f73369919217c21e4b68e6078f8aa872R207 ?

`cmrPassword` is the actual secret string, not the secret name, so the test creds are just using the secret directly? I don't see how the secret you're creating gets used

Reply:
The credentials (with the plaintext password) are used on line 231 to make sure that `publishECHO10XML2CMR()` is invoked with the correct parameters.

The actual secret is being used in that test because `reconcileCMRMetadata` -> `updateCMRMetadata` -> `getCreds`, which reads the secret from Secrets Manager. That password is then passed into `publish2CMR`.